lib/Dialect/Verif/VerifOps.cpp

```diff
@@ -58,6 +58,26 @@ LogicalResult LogicEquivalenceCheckingOp::verifyRegions() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// Formal Ops
+//===----------------------------------------------------------------------===//
+
+LogicalResult SymbolicValOp::verify() {
+  if (!isa<FormalOp>(getParentOp()))
+    return emitOpError()
+           << "Symbolic values can only be used in a formal test!";
```

Comment:
There is a `HasParent` constraint defined in tablegen that you could use to auto-generate this.

Reply:
Oh that's neat! I didn't know about that, thanks!